src/pools/D3MAaveV3TypePool.sol

```diff
@@ -0,0 +1,292 @@
+        //the current stable borrow rate. Expressed in ray
+        uint128 currentStableBorrowRate;
+        //timestamp of last update
+        uint40 lastUpdateTimestamp;
+        //the id of the reserve. Represents the position in the list of the active reserves
+        uint16 id;
+        //aToken address
+        address aTokenAddress;
+        //stableDebtToken address
+        address stableDebtTokenAddress;
+        //variableDebtToken address
+        address variableDebtTokenAddress;
+        //address of the interest rate strategy
+        address interestRateStrategyAddress;
+        //the current treasury balance, scaled
+        uint128 accruedToTreasury;
+        //the outstanding unbacked aTokens minted through the bridging feature
+        uint128 unbacked;
+        //the outstanding debt borrowed against this asset in isolation mode
+        uint128 isolationModeTotalDebt;
+    }
+    
+    function deposit(address asset, uint256 amount, address onBehalfOf, uint16 referralCode) external;
+    function withdraw(address asset, uint256 amount, address to) external;
+    function getReserveNormalizedIncome(address asset) external view returns (uint256);
+    function getReserveData(address asset) external view returns (ReserveData memory);
+    function flashLoanSimple(address receiverAddress, address asset, uint256 amount, bytes calldata params, uint16 referralCode) external;
+}
+
+interface RewardsClaimerLike {
+    function claimRewards(address[] calldata assets, uint256 amount, address to, address reward) external returns (uint256);
+}
+
+contract D3MAaveV3TypePool is ID3MPool {
+
+    mapping (address => uint256) public wards;
+    address                      public hub;
+    address                      public king; // Who gets the rewards
+    uint256                      public exited;
+
+    bytes32    public immutable ilk;
+    VatLike    public immutable vat;
+    PoolLike   public immutable pool;
+    ATokenLike public immutable stableDebt;
+    ATokenLike public immutable variableDebt;
+    ATokenLike public immutable adai;
+    TokenLike  public immutable dai; // Asset
+
+    uint256 internal constant FLASHLOAN_ENABLED_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFFF7FFFFFFFFFFFFFFF;
+    uint256 internal constant SUPPLY_CAP_MASK = 0xFFFFFFFFFFFFFFFFFFFFFFFFFF000000000FFFFFFFFFFFFFFFFFFFFFFFFFFFFF;
+    uint256 internal constant SUPPLY_CAP_START_BIT_POSITION = 116;
+
+    // --- Events ---
+    event Rely(address indexed usr);
+    event Deny(address indexed usr);
+    event File(bytes32 indexed what, address data);
+    event Collect(address indexed king, address indexed gift, uint256 amt);
+
+    constructor(bytes32 ilk_, address hub_, address dai_, address pool_) {
+        ilk = ilk_;
+        dai = TokenLike(dai_);
+        pool = PoolLike(pool_);
+
+        // Fetch the reserve data from Aave
+        PoolLike.ReserveData memory data = pool.getReserveData(dai_);
```

Comment:
in the aave pool and places were we access ilk/urn data we grab the specific variables we want rather than using a generic memory variable.  I think that is more explicit and cleaner as it avoids a bunch of `data.` references later

Reply:
There is stack too deep using the V2 method. I mention this in the top of the file.